plugins/engine-datafusion/src/main/java/org/opensearch/datafusion/search/cache/CacheManager.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.datafusion.search.cache;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.opensearch.common.settings.ClusterSettings;
+
+import static org.opensearch.datafusion.DataFusionQueryJNI.initCacheManagerConfig;
+
+
+public class CacheManager {
```

Comment:
Is this supposed to be thread safe? Do we really need a new impl for LFU/LRU caches?